### DISCUSSION TITLE
Manage custom resources in multiple deployments

### DISCUSSION THREAD

**Discussion by rublag** [Aidbox/Issues#488] [Q&A] [answered]
How to create multiple custom resources in `staging` environment and when they are ready for production create them in `production` environment?

**Comment by ier** [comment, marked as answer]
We recommend to use [Aidbox Project](https://docs.aidbox.app/aidbox-configuration/aidbox-zen-lang-project) for this.

## Example:

### Create custom resources with zen
#### `aidbox-project/resources/custom1.edn`:
```edn
{ns resources.custom1
 import #{zenbox}

 Custom1
 {:zen/tags #{zen/schema zenbox/persistent}
  :resourceType "Custom1"
  :confirms #{zenbox/Resource}
  :type zen/map
  :keys {:field1 {:type zen/string}}}}
```

#### `aidbox-project/resources/custom2.edn`:
```edn
{ns resources.custom2
 import #{zenbox}

 Custom2
 {:zen/tags #{zen/schema zenbox/persistent}
  :resourceType "Custom2"
  :confirms #{zenbox/Resource}
  :type zen/map
  :keys {:field1 {:type zen/string}}}}
```

### Create entrypoint for staging environment:
#### `aidbox-project/staging.edn`:
```edn
{ns staging
 import #{resources.custom1
          resources.custom2}}
```

### Create entrypoint for production environment
#### `aidbox-project/prod.edn`:
```edn
{ns prod
 import #{resources.custom1}}
```

### Specify Aidbox Project and entrypoint for staging environment:
```
AIDBOX_ZEN_PATHS:="path:dir:/path/to/aidbox-project"
AIDBOX_ZEN_ENTRYPOINT=staging
```

### Specify Aidbox Project and entrypoint for production environment:
```
AIDBOX_ZEN_PATHS:="path:dir:/path/to/aidbox-project"
AIDBOX_ZEN_ENTRYPOINT=staging
```

Now on production you will have `Custom1` resource and on staging `Custom1` and `Custom2` resources.
